### PR TITLE
Devel - CitCol Assembly Review

### PR DESCRIPTION
**Pull request by gkanogiannis**
Species: Citrullus colocynthis
Taxa: eudicots
Project: ERGA-Satellite

### CODE REVIEW

**Comment by diegomics** [comment]
Hi @gkanogiannis thanks for the PR! Review will start ASAP ;)

**Comment by diegomics** [comment]
Hi @gkanogiannis here are my initial comments:
1) Is this a hic-phased assembly? In that case, why is the hap2 missing? I do see the hap2 in the merqury kmer plot.
2) QV is missing from the yahs-scaffolded assembly. Has merqury generated this value? If yes, we need to understand why it is not being included in the report.
3) The hic contact map shows no discernible contacts. Can you provide a contact map without the scaffolds being painted?
4) What was the DNBSEQ data used for?
Thanks!

**Comment by gkanogiannis** [comment]
Hi @diegomics, thank you for your initial comments. Here are my responses. I also updated the pdf report.

> Hi @gkanogiannis here are my initial comments:
> 
> 1. Is this a hic-phased assembly? In that case, why is the hap2 missing? I do see the hap2 in the merqury kmer plot.
Yes, it is hic-phased assembly, but after contigging with hifiasm, hap2 was of bad quality and was discarded. We then proceeded with scaffolding and annotating only hap1. In the updated EAR pdf report I included hap2 contig report for reference.

> 2. QV is missing from the yahs-scaffolded assembly. Has merqury generated this value? If yes, we need to understand why it is not being included in the report.
Yes, merqury has generated files. You can see that kmer completeness values are correctly reported. Because we only scaffolded with hap1, merqury.qv has only one line for example "colocynth.asm.hic.hap1.p_ctg.chromosomes_final  1668    366467291       66.6406 2.16742e-07". I notice that `get_qv_value` function looks for "Both" and probably it fails with the single haplotype.

> 3. The hic contact map shows no discernible contacts. Can you provide a contact map without the scaffolds being painted?
Indeed the contact map picture is of bad quality, PretextView does not have an picture export functionality (does it??) so I had to take a screenshot. Any advice on how to produce a better picture?

> 4. What was the DNBSEQ data used for?
Initially I used short reads for genome profiling, but eventually hifi used for this as well. I was planning to use short reads for polishing, but again this was also skipped. So technically this data is unused, so I removed it from the updated report.

>    Thanks!

Thank you for your time. Looking forward for your next round of comments and hope I was able to answer your concerns. 

Anestis

**Comment by diegomics** [comment]
Try [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) (or HiGlass!)
For HiC phased assemblies, the usual proceeding is to scaffold both haplotypes (i.e., hap1 and hap2), to provide all the information available at the end.
I'll test the fix for the QV bug today ;)

**Comment by gkanogiannis** [comment]
> Try [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) (or HiGlass!) For HiC phased assemblies, the usual proceeding is to scaffold both haplotypes (i.e., hap1 and hap2), to provide all the information available at the end. I'll test the fix for the QV bug today ;)

Hi @diegomics. As a matter of fact, I did scaffolded both haps, but as I said, contig assembly of hap2 was bad, scaffold assembly of hap2 was also bad and hence I did not do any manual curation to it and basically forgot it existed :)  
Give me some time, I am running Merqury for both scaffolded hap1 and hap2 and will update EAR pdf.

Thanks for the hint about PretextSnapshot, I will give a try and let you know if I can update further the pdf with a better image.

Cheers,

Anestis

**Comment by gkanogiannis** [comment]
Hi @diegomics and @tbrown91. Although the improved `get_qv_value` works fine now for parsing single hap merqury result, I updated my report to include both scaffolded haps.

Anestis

**Comment by tbrown91** [comment]
Hi Anestis,

You're right that the hap2 is a bit worrying. It looks like the phasing has not worked at all well in these homozygous regions, which is perhaps expect for such a low heterozygosity genome. It looks like there is still a fair amount of retained haplotypes in both assemblies, did you run purge-dups after hifiasm? I suspect that this would remove only these 2-copy contigs that are visible in the merqury plot. If these sequences were not incorporated into scaffolds, hopefully you can run purge-dups on your final assembly and see.

I don't think the smaller hap2 genome will be a big problem come submission time. I think it important that you include both assemblies and label the other as an "alternate pseudo-haplotype".

Tom

**Comment by gkanogiannis** [comment]
> Hi Anestis,
> 
> You're right that the hap2 is a bit worrying. It looks like the phasing has not worked at all well in these homozygous regions, which is perhaps expect for such a low heterozygosity genome. It looks like there is still a fair amount of retained haplotypes in both assemblies, did you run purge-dups after hifiasm? I suspect that this would remove only these 2-copy contigs that are visible in the merqury plot. If these sequences were not incorporated into scaffolds, hopefully you can run purge-dups on your final assembly and see.
> 
> I don't think the smaller hap2 genome will be a big problem come submission time. I think it important that you include both assemblies and label the other as an "alternate pseudo-haplotype".
> 
> Tom

Hi @tbrown91, thank you for your time reviewing my assembly.

I did not run purge_dups separately, but I used the purge-dups functionality of hifiasm (-l1). I update the report again, where I include the contig assemblies with purge_dups.
You will notice that hap1 is very similar for the two modes, but hap2 is better for the purged case. At the end I had to stick with one for annotation, and selected the non purged hap1, as I was able to get 2 more telomeres :)
I will try to update later the data for the scaffolded haps for the purged case (currently you can only see the data for the contig level).

Cheers,

Anestis

**Comment by diegomics** [comment]
Hi Anestis,

If I understand correctly, you are showing in the contigging section of the report, two different assemblies: one produced with Hifiasm without purging duplication options, and one produced with -l1 option enabled. Am I correct? The report should reflect a linear proceeding, from contigs to scaffolds.

Running purge_dups is independent of the bundled purging options of Hifiasm. Sometimes it is not needed at all (we more often see this in Hifiasm HiC-phased assemblies for vertebrate genomes), but it depends a lot on the genome, on the dataset, etc

**Comment by gkanogiannis** [comment]
Hi Diego,

> Hi Anestis,
> 
> If I understand correctly, you are showing in the contigging section of the report, two different assemblies: one produced with Hifiasm without purging duplication options, and one produced with -l1 option enabled. Am I correct? The report should reflect a linear proceeding, from contigs to scaffolds.
> 

Yes, exactly. The contig assemblies I originally reported re with -l0 (no purging) and is the ones I stick with for scaffolding and further annotation. 


> Running purge_dups is independent of the bundled purging options of Hifiasm. Sometimes it is not needed at all (we more often see this in Hifiasm HiC-phased assemblies for vertebrate genomes), but it depends a lot on the genome, on the dataset, etc

So, if I understand correctly, it is better to completely remove from the report the assemblies with the purging option from hifiasm? To be clear, I have not used `purge_dups` at all.

Cheers,

Anestis

**Comment by diegomics** [comment]
Ok, then Hifiasm-purge_dups is an incorrect naming, but more importantly, that should not be included since YaHS results are based on the Hifiasm -l0 run.

It seems that Hifiasm -l0 is not balancing correctly the haplotypes. This is improved in the -l1 run. It would be interesting to try purge_dups on both the -l0 and -l1 runs to check which produces the better assembly. What do you think @tbrown91 ? Did you also run with the --primary to obtain pri and alt in addition to hap1 and hap2?

**Comment by gkanogiannis** [comment]
Hi @diegomics and @tbrown91,

> Ok, then Hifiasm-purge_dups is an incorrect naming, but more importantly, that should not be included since YaHS results are based on the Hifiasm -l0 run.
> 

I agree, so I update again the report and completely remove the contig assemblies from hifiasm with purging.

> It seems that Hifiasm -l0 is not balancing correctly the haplotypes. This is improved in the -l1 run. It would be interesting to try purge_dups on both the -l0 and -l1 runs to check which produces the better assembly. What do you think @tbrown91 ? Did you also run with the --primary to obtain pri and alt in addition to hap1 and hap2?

I did run hifiasm with -l0 and -l1 and with --primary. But I did not run purge_dups separately. After the initial evaluation of those contig assemblies, I decided to go with the hap1 without purging, since the genome was not so complex and was happy with it, in terms of quality. 

Unfortunately, I am heavily understaffed here (well, bioinfo team is just me) and the computational resources very scarce. It took me months to convince the management that we need to replace the shitty cluster with a modern HPC, but this will come in 3-4 months. I don't really have the luxury at the moment to try purge_dups, because I have already finished the annotation and it took weeks to run :( 

On the other hand, I have another plant that I am going to work with, we already have the sequencing data. This is more complex, initial profiling shows crazy level of repeatition and heterozygocity and I will have the opportunity with it to try many more assembling and purging options before I freeze the assembly and go for annotation.

Anestis

**Comment by diegomics** [comment]
Please consider using [Galaxy](usegalaxy.eu) if resources are a limiting factor. You can easily run PretextSnapshot there for example. Since this is a 'Hifiasm' assembly, please take a look at the available workflows, there is a nice one for purge_dups too ;)

**Comment by gkanogiannis** [comment]
> Please consider using [Galaxy](usegalaxy.eu) if resources are a limiting factor. You can easily run PretextSnapshot there for example. Since this is a 'Hifiasm' assembly, please take a look at the available workflows, there is a nice one for purge_dups too ;)

Thanks Diego, surely I will consider these for the plant species I am working on the next months.
For C.colocynthis, unfortunately I don't have the time for further analysis and I am pressed from above for releasing.
Isn't the current assembly (hifiasm -l0, no purge_dups, yahs) of satisfactory quality for EAR?

Cheers,

Anestis

**Comment by tbrown91** [comment]
Hi Anestis,

We wouldn't normally recommend using -l 0, as it leaves a number of haplotigs in the assembly that should be purged, and in your case seems to have produced very unbalanced haplotypes. We can't really make a final call until we see what the HiC map looks like. From what you've included it's very difficult to tell if there is any HiC signal at all.

If you are going ahead with submission, you can paste the BioProject here. I recommend having a look at the Sample Manifest from the Sample and Sample Processing Committee to ensure you are uploading sufficient metadata with your sample.

Tom

**Comment by gkanogiannis** [comment]
Hi Tom,

Thank you for your response and comments.

> Hi Anestis,
> 
> We wouldn't normally recommend using -l 0, as it leaves a number of haplotigs in the assembly that should be purged, and in your case seems to have produced very unbalanced haplotypes. We can't really make a final call until we see what the HiC map looks like. From what you've included it's very difficult to tell if there is any HiC signal at all.
> 

You have a point and actually I am considering to spare some time and go back and redo the annotation with the hifiasm purged (-l1) assembly. You are also very right that the contact map picture looks weird. In fact, I had done the scaffolding with the juicer/juicebox/3d-dna pipeline as well, and the map looks like the attached picture. What do you think? Do you think there is an issue with the Hi-C data? 
out_JBAT JB


> If you are going ahead with submission, you can paste the BioProject here. I recommend having a look at the Sample Manifest from the Sample and Sample Processing Committee to ensure you are uploading sufficient metadata with your sample.
> 

I am trying to contact the SSP committee on how to create the BioSample. Actually, I have already created it, by concatenating the ENA Plant Sample checklist with all the mandatory fields from the SSP manifest (2.3.4), but many values are "noy_applicable" and wanted to validate it with them.

> Tom

Anetis

**Comment by tbrown91** [comment]
Hi Anestis that looks much better. I'm pretty happy with the scaffolding now, good job.

If you want contact details for someone from the SSP, let me know and I will send you some email addresses privately, so as not to broadcast them for all on the internet

**Comment by gkanogiannis** [comment]
Dear @diegomics and @tbrown91 . I updated the EAR report. This time I used hifiasm with light purging (-l1) and scaffolded both haps with yahs. Contact maps are from JuiceBox, as the ones from Pretext don't show very well.

Anestis

**Comment by diegomics** [comment]
Hi @gkanogiannis thanks for the updates. I will ask you for one more small fix, and then I'm more than happy to approve this to go for manual curation.

Would it be possible to run the make_EAR.py script again? I just did a quick update to correct the naming of the hic contact map title. In your case, you are not using pretext, so now it will use a generic name. In addition, would it be possible for you to use a better-quality PNG for both contact maps? Thanks for all the hard work.

**Comment by tbrown91** [comment]
Hi @gkanogiannis This is looking pretty nice. A couple of points from me:

It looks like the haploid number listed on GoaT is incorrect and there are indeed 11 chromosomes per haplotype in this species.

I'm not sure if there are a few retained haplotypes in hap2 (small blue curve under the main peak), but I don't think there are any red flags from the busco scores. The hap1 assembly looks like a very good primary assembly.

If I read this correctly, this is the HiC map after running yahs with no curation, which is pretty amazing. It looks like only the last two visible scaffolds in hap2 need to be joined, but maybe if you zoom in a bit closer there will be more.

In short: This looks like a very nice haplotype-phased genome with little to change.

**Comment by gkanogiannis** [comment]
> Hi @gkanogiannis thanks for the updates. I will ask you for one more small fix, and then I'm more than happy to approve this to go for manual curation.
> 
> Would it be possible to run the make_EAR.py script again? I just did a quick update to correct the naming of the hic contact map title. In your case, you are not using pretext, so now it will use a generic name. In addition, would it be possible for you to use a better-quality PNG for both contact maps? Thanks for all the hard work.

Hi @diegomics, I updated the report and tried to get better map exports from Juicebox.

Anestis

**Comment by gkanogiannis** [comment]
> Hi @gkanogiannis This is looking pretty nice. A couple of points from me:
> 
> It looks like the haploid number listed on GoaT is incorrect and there are indeed 11 chromosomes per haplotype in this species.
> 
> I'm not sure if there are a few retained haplotypes in hap2 (small blue curve under the main peak), but I don't think there are any red flags from the busco scores. The hap1 assembly looks like a very good primary assembly.
> 
> If I read this correctly, this is the HiC map after running yahs with no curation, which is pretty amazing. It looks like only the last two visible scaffolds in hap2 need to be joined, but maybe if you zoom in a bit closer there will be more.
> 
> In short: This looks like a very nice haplotype-phased genome with little to change.

Hi @tbrown91. Yes indeed, there are 11 chromosomes per haplotype (it's a wild relative of watermelon which has same number). I also think hap1 looks ok and have started the annotation tasks with it.

Anestis

**Review by diegomics — APPROVED** [review]
All looking good. As @tbrown91 pointed out, manual curation here should be pretty straightforward. Thanks!

**Review by tbrown91 — APPROVED** [review]

**Comment by gkanogiannis** [comment]
Dear @diegomics and @tbrown91, thank you so much for your time and the great work for the ERGA community.

Anestis
